app/models/user.rb

```diff
@@ -3,16 +3,40 @@ class User < ActiveRecord::Base
     config.authentications_class = Authentication
   end
 
-  has_many :authentications, dependent: :destroy
-  accepts_nested_attributes_for :authentications
+  after_create :add_current_deck
 
-  validates :password, length: { minimum: 3 }
-  validates :password, confirmation: true
-  validates :password_confirmation, presence: true
+  belongs_to :current_deck, class_name: "Deck"
+  has_many :decks
+  has_many :authentications, dependent: :destroy
+  has_many :cards, through: :decks
 
+  with_options :if => lambda {|obj| puts obj.password.empty?} do |u|
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Use the new lambda literal syntax `->(params) {...}`.<br>Space between { and | missing.<br>Space missing inside }.<br>Do not write to stdout. Use Rails' logger if you want to log.